phc/offset.go

```diff
@@ -102,21 +105,60 @@ func TimeAndOffsetFromDevice(device string, method TimeMethod) (SysoffResult, er
 			return SysoffResult{}, fmt.Errorf("failed clock_gettime: %w", err)
 		}
 
-		return sysoffEstimateBasic(ts1, time.Unix(ts.Unix()), ts2), nil
+		return SysoffEstimateBasic(ts1, time.Unix(ts.Unix()), ts2), nil
 	case MethodIoctlSysOffsetExtended:
-		extended, err := ReadPTPSysOffsetExtended(device, 5)
+		extended, err := ReadPTPSysOffsetExtended(device, ExtendedNumProbes)
 		if err != nil {
 			return SysoffResult{}, err
 		}
-		return sysoffEstimateExtended(extended), nil
+		return SysoffEstimateExtended(extended), nil
 	}
 	return SysoffResult{}, fmt.Errorf("unknown method to get PHC time %q", method)
 }
 
-func CalcPHCOffet(timeAndOffsetA, timeAndOffsetB SysoffResult) (PHCDiff time.Duration) {
-	sysOffset := timeAndOffsetB.SysTime.Sub(timeAndOffsetA.SysTime)
-	phcOffset := timeAndOffsetB.PHCTime.Sub(timeAndOffsetA.PHCTime)
-	phcOffset -= sysOffset
+// generics in standard library can't come soon enough...
+func abs(value time.Duration) time.Duration {
+	if value < 0 {
+		return -value
+	}
+	return value
+}
+
+// OffsetBetweenExtendedReadings returns estimated difference between two PHC SYS_OFFSET_EXTENDED readings
+func OffsetBetweenExtendedReadings(extendedA, extendedB *PTPSysOffsetExtended) time.Duration {
+	// we expect both probes to have same number of measures
+	numProbes := int(extendedA.NSamples)
```

Comment:
What if it's not the same number?

Reply:
We can of course add error to the signature and return error in such case, but I thought it's not worth it.